matplotcheck/base.py

```diff
@@ -921,20 +918,127 @@ def assert_xydata(
             two datasets because it is able to account for small errors in
             floating point numbers, and it scales nicely between extremely
             small or large numbers. We catch this error and throw our own so
-            that we can use our own message."""
-            try:
-                np.testing.assert_array_max_ulp(
-                    np.array(xy_data["x"]), np.array(xy_expected[xcol])
-                )
-            except AssertionError:
-                raise AssertionError(message)
+            that we can use our own message. We can't use this to compare
+            datetime.datetime objects, so we use our own comparison."""
+
+            # Compare x-values
+            if xtime:
+                if not self._compare_time(xy_data["x"], xy_expected[xcol]):
+                    raise AssertionError(message)
+            else:
+                try:
+                    np.testing.assert_array_max_ulp(
+                        np.array(xy_data["x"]), np.array(xy_expected[xcol])
+                    )
+                except AssertionError:
+                    raise AssertionError(message)
+
+            # compare y-values
             try:
                 np.testing.assert_array_max_ulp(
-                    np.array(xy_data["y"]), np.array(xy_expected[ycol])
+                    xy_data["y"].values.tolist(),
+                    xy_expected[ycol].values.tolist(),
                 )
             except AssertionError:
                 raise AssertionError(message)
 
+    def _compare_time(self, time_1, time_2):
```

Comment:
I created a _compare_time function to handle time comparisons between matplotlib's format and `pandas.timestamp`. This cleans up `assert_xydata()` and allows this code to be reused elsewhere.